src/plugins/charts/public/services/palettes/helpers.ts

```diff
@@ -30,25 +31,55 @@ function findColorsByStops(
   stops: number[]
 ) {
   const index = stops.findIndex((s) => comparison(value, s) < 0);
-  return colors[index] || colors[0];
+  return colors[index] || colors[colors.length - 1];
 }
 
 function getNormalizedValueByRange(
   value: number,
   { range }: CustomPaletteState,
-  minMax: { min: number; max: number }
+  minMax: { min: number; max: number },
+  isMaxContinuity: boolean,
+  rangeMax: number
 ) {
   let result = value;
   if (range === 'percent') {
     result = (100 * (value - minMax.min)) / (minMax.max - minMax.min);
   }
   // for a range of 1 value the formulas above will divide by 0, so here's a safety guard
   if (Number.isNaN(result)) {
-    return 1;
+    return isMaxContinuity ? 100 : rangeMax;
```

Comment:
Not sure what is the exact meaning of this: 100 should be only for `percent` type, no?

Reply:
according to the comment on 48 line that if should work only after formula on 46 line, that formula execute only for percent mode. I moved that 'if' inside if statement on 45 line